utils/helpers/formatting/formatting.ts

```diff
@@ -9,15 +10,22 @@ const timeZone = 'UTC';
  * @returns
  */
 const safe =
-  (castAndFormat: Function) =>
-  (...args: any) => {
+  <T extends unknown[], R>(castAndFormat: (...args: T) => R) =>
+  (...args: T): R | string => {
     try {
       return castAndFormat.apply(null, args);
     } catch (e: any) {
-      logErrorInSentry('Error while using formatting function', {
-        details: e.toString() || args,
-      });
-      return args;
+      logErrorInSentry(
+        new InternalError({
+          message: 'Formatting error in view',
+          cause: e,
+          context: {
+            details: args.toString(),
+          },
+        })
+      );
+
+      return args.toString();
```

Comment:
cant args be undefined ?

If so it will return a typeError

Reply:
Nop, it can't. A function always have args (an array of arg). 
You can try it at home : 
```js
(function(...args) {console.log(args.toString())})()
```

Reply:
Nop, it can't. A function always have args (an array of arg). 
You can try it at home : 
```js
(function(...args) {console.log(args.toString())})()
```

Reply:
but some type can be problematic, though :
ex `symbol(4)`

I can check for that

Reply:
Done ! It should be bulletproof now :)